### PR TITLE
APP-156 Refactor getFeeLevels and fix tests

### PR DESCRIPTION
**Pull request by gonzaloleon**

### CODE REVIEW

**Review by cmgustavo — APPROVED** [review]
ACK! 👍🏼

File: src/components/choose-fee-level/choose-fee-level.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export class ChooseFeeLevelComponent extends ActionSheetParent {
 
     this.loadingFee = true;
     this.feeProvider
-      .getFeeLevels(this.coin)
+      .getFeeLevels(this.coin, this.network)
       .then(levels => {
```
Comment (matiu) [inline src/components/choose-fee-level/choose-fee-level.ts:89]: Maybe change "levels" to "response" so....

File: src/components/choose-fee-level/choose-fee-level.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export class ChooseFeeLevelComponent extends ActionSheetParent {
           );
           return;
         }
-        this.feeLevels = levels;
+        this.feeLevels = levels.levels;
```
Comment (matiu) [inline src/components/choose-fee-level/choose-fee-level.ts:98]: ... here is. "response.levels".

File: src/components/choose-fee-level/choose-fee-level.ts [hunk 3]
```diff
@@ -118,7 +118,7 @@ export class ChooseFeeLevelComponent extends ActionSheetParent {
   }
 
   public setFeeRates() {
-    this.feeLevels.levels[this.network].forEach((feeLevel, i) => {
+    this.feeLevels.forEach((feeLevel, i) => {
```
Comment (matiu) [inline src/components/choose-fee-level/choose-fee-level.ts:121]: nice!

File: src/providers/fee/fee.ts [hunk 4]
```diff
@@ -103,46 +109,46 @@ export class FeeProvider {
     });
   }
 
-  public getFeeLevels(coin: string): Promise<any> {
+  public getFeeLevels(coin: string, network: string): Promise<any> {
     return new Promise((resolve, reject) => {
       coin = coin || 'btc';
-
+      const indexFound = this.cache.findIndex(
+        fl => fl.coin == coin && fl.network == network
```
Comment (matiu) [inline src/providers/fee/fee.ts:116]: 👍

**Comment by matiu** [comment]
ACK! Great job, much better than before

**Review by cmgustavo — APPROVED** [review]

**Review by Gamboster — APPROVED** [review]

**Pull request merged**